tests/python/unittest/test_custom_datatypes.py

```diff
@@ -0,0 +1,396 @@
+    kshape = (32, 4, 3, 3)
+    run_test_conv2d(src_dtype,
+                    dst_dtype,
+                    1,
+                    dshape,
+                    kshape,
+                    padding=(1, 1),
+                    channels=32,
+                    groups=8,
+                    kernel_size=(3, 3))
+    # also group conv2d
+    dshape = (1, 32, 18, 18)
+    kshape = (64, 1, 3, 3)
+    run_test_conv2d(src_dtype,
+                    dst_dtype,
+                    1,
+                    dshape,
+                    kshape,
+                    padding=(1, 1),
+                    channels=64,
+                    groups=32,
+                    kernel_size=(3, 3))
+
+    # normal conv2d
+    dshape = (1, 3, 224, 224)
+    kshape = (10, 3, 3, 3)
+    run_test_conv2d(src_dtype,
+                    dst_dtype,
+                    1,
+                    dshape,
+                    kshape,
+                    padding=(1, 1),
+                    channels=10,
+                    kernel_size=(3, 3))
+
+    # dilated conv2d
+    dshape = (1, 3, 18, 18)
+    kshape = (10, 3, 3, 3)
+    run_test_conv2d(src_dtype,
+                    dst_dtype,
+                    1,
+                    dshape,
+                    kshape,
+                    padding=(1, 1),
+                    channels=10,
+                    kernel_size=(3, 3),
+                    dilation=(3, 3))
+
+
+def test_ops():
+    run_ops('float32', 'custom[posites2]8', rtol=1, atol=1)
+    run_ops('float32', 'custom[posites2]16', rtol=0.01, atol=1)
+    run_ops('float32', 'custom[posites2]32')
+
+def test_conv2d():
+    run_conv2d('float32', 'custom[posites2]8', rtol=1, atol=1)
+    run_conv2d('float32', 'custom[posites2]16', rtol=0.01, atol=1)
+    run_conv2d('float32', 'custom[posites2]32')
+
+def test_batchnorm():
+    def run_batchnorm(src_dtype, dst_dtype, rtol=1e-4, atol=1e-4):
+        shape = (3, 32, 32)
+        t = relay.TensorType(shape, src_dtype)
+        x = relay.var("x", t)
+        bn = batch_norm_infer(data=x, epsilon=2e-5, scale=False, name='bn_x')
+        f = relay.Function(relay.analysis.free_vars(bn), bn)
+
+        x_data = rs.rand(*shape).astype(t.dtype)
+        module = tvm.IRModule.from_expr(f)
+
+        zero_data = np.zeros((32), 'float32')
+        compare(module, (x_data, zero_data, zero_data, zero_data, zero_data), src_dtype, dst_dtype, rtol, atol)
+
+    run_batchnorm('float32', 'custom[posites2]8', rtol=1, atol=1)
+    run_batchnorm('float32', 'custom[posites2]16', rtol=0.01, atol=1)
+    run_batchnorm('float32', 'custom[posites2]32')
+
+def test_models():
+    # Expected posit8 might be faster, but it's not.
+    # run_model(get_mobilenet, (3, 224, 224), 'float32', 'custom[posit8]8')
+    # run_model(get_mobilenet, (3, 224, 224), 'float32', 'custom[posit32]32')
+    # run_model(get_inception, (3, 299, 299), 'float32', 'custom[posit32]32')
+    # run_model(get_resnet, (3, 224, 224), 'float32', 'custom[posit32]32')
```

Comment:
can we remove these?

Reply:
Let's talk about it -- I think we should at least leave a few around, to document which models are too slow to run. We can move them down, though.